src/app/games/sahara/sahara.component.ts

```diff
@@ -66,13 +68,13 @@ export class SaharaComponent extends TriangularGameComponent<SaharaMove, SaharaP
             } catch (error) {
                 return this.cancelMove(error.message);
             }
-            return await this.chooseMove(newMove, this.rules.node.gamePartSlice, null, null);
+            return await this.chooseMove(newMove, this.rules.node.gameState, null, null);
         }
     }
     private choosePiece(x: number, y: number): MGPValidation {
-        if (this.board[y][x] === FourStatePiece.EMPTY.value) { // Did not select pyramid
+        if (this.board[y][x] === FourStatePiece.EMPTY) { // Did not select pyramid
             return this.cancelMove(SaharaFailure.MUST_CHOOSE_PYRAMID_FIRST);
-        } else if (this.getTurn() % 2 === this.board[y][x]) { // selected his own pyramid
+        } else if (this.getTurn() % 2 === this.board[y][x].value) { // selected his own pyramid
```

Comment:
this.getTurn() % 2 c'est this.getCurrentPlayer() non ? et du coup pas besoin de .value à droite